apps/omg_child_chain/test/omg_child_chain/datadog_event/contract_event_consumer_test.exs

```diff
@@ -41,22 +41,23 @@ defmodule OMG.ChildChain.DatadogEvent.ContractEventConsumerTest do
     :ok
   end
 
-  test "if a event message put on omg bus is consumed by the event consumer and published on the publisher interface" do
+  test "if an event message put on omg bus is consumed by the event consumer and published on the publisher interface" do
     topic_name = self() |> :erlang.pid_to_list() |> to_string()
     sig = "#{topic_name}(bytes32)"
     data = [%{event_signature: sig}]
     {:root_chain, topic_name} |> OMG.Bus.Event.new(:data, data) |> OMG.Bus.direct_local_broadcast()
-
     assert_receive {:event, _, _}
   end
 
-  test "if a list of event message are put on omg bus is consumed by the event consumer and published on the publisher interface" do
+  test "if a list of events put on omg bus is consumed by the event consumer, broken down and published individually on the publisher interface" do
     topic_name = self() |> :erlang.pid_to_list() |> to_string()
     sig = "#{topic_name}(bytes32)"
     data = [%{event_signature: sig}, %{event_signature: sig}]
     {:root_chain, topic_name} |> OMG.Bus.Event.new(:data, data) |> OMG.Bus.direct_local_broadcast()
 
-    assert_receive {:event, _, _}
+    Enum.each(data, fn _ ->
+      assert_receive {:event, _, _}
```

Comment:
Now that it expect 2 events, should we check that they're indeed the two events, and not one event reported twice?

Reply:
Absolutely, i'm taking the message into equation then